core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -222,65 +222,72 @@ public PGConnectionContext(
-        this.namedStatementMap = new CharSequenceObjHashMap<>(configuration.getNamedStatementCacheCapacity());
-        this.pendingWriters = new ObjObjHashMap<>(configuration.getPendingWritersCacheSize());
-        this.namedPortalMap = new CharSequenceObjHashMap<>(configuration.getNamedStatementCacheCapacity());
-        this.binarySequenceParamsPool = new ObjectPool<>(DirectBinarySequence::new, configuration.getBinParamCountCapacity());
-
-        this.metrics = engine.getMetrics();
-        final boolean enableSelectCache = configuration.isSelectCacheEnabled();
-        final int selectBlockCount = enableSelectCache ? configuration.getSelectCacheBlockCount() : 1;
-        final int selectRowCount = enableSelectCache ? configuration.getSelectCacheRowCount() : 1;
-        this.typesAndSelectCache = new AssociativeCache<>(
-                selectBlockCount,
-                selectRowCount,
-                metrics.pgWire().cachedSelectsGauge(),
-                metrics.pgWire().selectCacheHitCounter(),
-                metrics.pgWire().selectCacheMissCounter()
-        );
-        this.typesAndSelectPool = new WeakSelfReturningObjectPool<>(TypesAndSelect::new, selectBlockCount * selectRowCount);
-
-        final boolean enabledUpdateCache = configuration.isUpdateCacheEnabled();
-        final int updateBlockCount = enabledUpdateCache ? configuration.getUpdateCacheBlockCount() : 1;
-        final int updateRowCount = enabledUpdateCache ? configuration.getUpdateCacheRowCount() : 1;
-        this.typesAndUpdateCache = new AssociativeCache<>(
-                updateBlockCount,
-                updateRowCount,
-                metrics.pgWire().cachedUpdatesGauge()
-        );
-        this.typesAndUpdatePool = new WeakSelfReturningObjectPool<>(parent -> new TypesAndUpdate(parent, engine), updateBlockCount * updateRowCount);
-
-        final boolean enableInsertCache = configuration.isInsertCacheEnabled();
-        final int insertBlockCount = enableInsertCache ? configuration.getInsertCacheBlockCount() : 1;
-        final int insertRowCount = enableInsertCache ? configuration.getInsertCacheRowCount() : 1;
-        this.typesAndInsertCache = new AssociativeCache<>(insertBlockCount, insertRowCount);
-        this.typesAndInsertPool = new WeakSelfReturningObjectPool<>(TypesAndInsert::new, insertBlockCount * insertRowCount);
 
-        this.batchCallback = new PGConnectionBatchCallback();
-        this.queryTag = TAG_OK;
-        this.queryContainsSecret = false;
-        FactoryProvider factoryProvider = configuration.getFactoryProvider();
-        this.securityContextFactory = factoryProvider.getSecurityContextFactory();
+        try {
+            this.path = new Path();
+            this.engine = engine;
+            queryLogger = engine.getConfiguration().getQueryLogger();
+            this.utf8Sink = new DirectUtf16Sink(engine.getConfiguration().getTextConfiguration().getUtf8SinkSize());
```

Comment:
is this a utf8 or utf16 sink? or is this a trick? :)

Reply:
I deleted the entire field, nothing seems to be using it.